src/celutil/formatnum.cpp

```diff
@@ -146,8 +159,66 @@ parseExponential(fmt::memory_buffer& buffer, ParseExpResult& result)
     return ec == std::errc{} && ptr == exponentEnd;
 }
 
+#ifdef USE_ICU
+bool getNumberSymbol(const UNumberFormat *numFormat, UNumberFormatSymbol symbol, std::string &output)
+{
+    UErrorCode status = U_ZERO_ERROR;
+    auto requiredSize = unum_getSymbol(numFormat, symbol, nullptr, 0, &status);
+    if (U_FAILURE(status) && status != U_BUFFER_OVERFLOW_ERROR)
+        return false;
+
+    status = U_ZERO_ERROR;
+    std::u16string symbolString;
+    symbolString.resize(requiredSize);
+    unum_getSymbol(numFormat, symbol, symbolString.data(), requiredSize, &status);
+
+    if (U_FAILURE(status))
+        return false;
+
+    u_strToUTF8(nullptr, 0, &requiredSize, symbolString.data(), symbolString.size(), &status);
+    if (U_FAILURE(status) && status != U_BUFFER_OVERFLOW_ERROR)
+        return false;
+
+    status = U_ZERO_ERROR;
+    output.resize(requiredSize);
+
+    u_strToUTF8(output.data(), requiredSize, nullptr, symbolString.data(), symbolString.size(), &status);
+    return U_SUCCESS(status);
+}
+#endif
+
 } // end unnamed namespace
 
+#ifdef USE_ICU
+NumberFormatter::NumberFormatter()
+{
+    const char *orig = N_("LANGUAGE");
```

Comment:
I don't think we should use this mechanism - should pass `nullptr` as the locale to specify the default locale.

Reply:
changed to use nullptr instead